### PR TITLE
SAK-50341 Lessons missing condition picker when editing text on a subpage

### PR DESCRIPTION
**Pull request by jesusmmp**
https://sakaiproject.atlassian.net/browse/SAK-50341

### CODE REVIEW

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/util/LessonConditionUtil.java [hunk 1]
```diff
@@ -36,12 +36,16 @@ public static UIComponent makeConditionEditor(SimplePageBean simplePageBean, UIC
     public static UIComponent makeConditionPicker(SimplePageBean simplePageBean, UIContainer parent, String rsfId) {
```
Comment (stetsche) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/util/LessonConditionUtil.java:36]:
```suggestion
    // Make condition picker based on current item in page bean
    public static UIComponent makeConditionPicker(SimplePageBean simplePageBean, UIContainer parent, String rsfId) {
```

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/util/LessonConditionUtil.java [hunk 2]
```diff
@@ -36,12 +36,16 @@ public static UIComponent makeConditionEditor(SimplePageBean simplePageBean, UIC
...
-                ? UIOutput.make(parent, rsfId)
-                        .decorate(new UIFreeAttributeDecorator("site-id", simplePageBean.getCurrentSiteId()))
-                        .decorate(new UIFreeAttributeDecorator("tool-id", LessonsEntityProvider.TOOL_COMMON_ID))
-                        .decorate(new UIFreeAttributeDecorator("lesson-id", String.valueOf(currentPageItem.getId())))
-                : null;
-   }
+        return makeConditionPicker(currentPageItem, simplePageBean.getCurrentSiteId(), parent, rsfId);
+    }
+
+    public static UIComponent makeConditionPicker(SimplePageItem simplePageItem, String siteId, UIContainer parent, String rsfId) {
```
Comment (stetsche) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/util/LessonConditionUtil.java:42]:
```suggestion
    // Make condition picker based on a specific page item
    public static UIComponent makeConditionPicker(SimplePageItem simplePageItem, String siteId, UIContainer parent, String rsfId) {
```

**Review by stetsche — APPROVED** [review]
LGTM, thanks @jesusmmp! Maybe you could add this comments :)

**Review by bgarciaentornos — APPROVED** [review]
Nice, thank you both!

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/FolderPickerProducer.java [hunk 3]
```diff
@@ -87,9 +87,10 @@ public void fillComponents(UIContainer tofill, ViewParameters viewparams, Compon
         Long itemId = ((GeneralViewParameters) viewparams).getItemId();
         simplePageBean.setItemId(itemId);
         
+        SimplePageItem i = null;
```
Comment (ern) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/FolderPickerProducer.java:90]: i is probably not a very good variable name in this context!
```        SimplePageItem item = null;
```

**Comment by jesusmmp** [comment]
It doesn't work!

**Pull request closed without merging**